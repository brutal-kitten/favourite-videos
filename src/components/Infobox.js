import React, { Component } from 'react';
import PropTypes from 'prop-types'

class Infobox extends Component {

  static propTypes = {
    title: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired,
    likes: PropTypes.string.isRequired,
    views: PropTypes.string.isRequired
  }


  render() {



    return (
      <div className="infobox grid-item">
        <div className="title"> Title: <span>{this.props.title}</span></div>
        <div className="addedAt">Published At: <span> {this.props.date}</span></div>
        <div className="likes"><span>{this.props.likes}</span> Likes</div>
        <div className="views"><span>{this.props.views}</span> Views</div>
      </div>
    )
  }

}

export default Infobox;
