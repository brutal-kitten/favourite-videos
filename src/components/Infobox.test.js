import React from 'react';
import ReactDOM from 'react-dom';
import Infobox from './Infobox.js';
import TestRenderer from 'react-test-renderer';
import { shallow, mount, render, ReactWrapper } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


configure({ adapter: new Adapter() });

it('renders properly', () => {
  const tree = TestRenderer.create(
    <Infobox />).toJSON();
  expect(tree).toMatchSnapshot();
});

const props = {
  title: "name",
  date: new Date("2017-05-31T09:30:02.000Z").toDateString(),
  likes: '8359285',
  views: '349487'
}

const wrapper = mount(<Infobox {...props}/>);

describe('should have given given props', () =>  {

  it('should have given title', () =>  {
    expect(wrapper.prop('title')).toBe(props.title);
  });

  it('should have given date', () =>  {
    expect(wrapper.prop('date')).toBe(props.date);
  });

  it('should have given likes', () =>  {
    expect(wrapper.prop('likes')).toBe(props.likes);
  });

  it('should have given views', () =>  {
    expect(wrapper.prop('views')).toBe(props.views);
  });
});
