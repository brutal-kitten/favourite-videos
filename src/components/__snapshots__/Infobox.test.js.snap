// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<div
  className="info"
>
  <div
    className="title infobox"
  />
  <div
    className="addedAt infobox"
  />
  <div
    className="likes infobox"
  >
    <span />
     
    <span
      className="glyphicon glyphicon-heart"
    />
  </div>
  <div
    className="views infobox"
  >
    <span />
     views
  </div>
</div>
`;
